test-data/unit/check-newsemanal.test

```diff
@@ -1580,6 +1580,40 @@ class C:
 def C() -> None:  # E: Name 'C' already defined on line 1
     ''()  # E: "str" not callable
 
+[case testNewAnalyzerTupleIteration]
+from typing import Union, Tuple, NamedTuple
+
+class A: pass
+class B(A): pass
+class C(A): pass
+
+class T(Tuple[B, C]):
+    pass
```

Comment:
Maybe put the classes in reverse order (just to check that forward refs also work, although I am quite they should).